File: LanguageTranslator/tests/C/while.c
/*
 * University of Glasgow
 * Msc Project fall, 2019
 * Author: Yidi Cao
 * Acknowledgement:
 * This file is partly based on the Fun Compiler at https://moodle.gla.ac.uk/course/view.php?id=903
 * 
*/
//Test while-commands
int pos(int a){
	return a;
}

void main(){
	int a ;
	scanf(a);
	int i = 1;
	char c = 's';
	while(i*i < a+1){ //condition with comparison&arithmatic expression
		printf(i*i); //print function called in while statement
		i = i+1; //assignmeng command in while 
	}
	while(c != 'a'){ //condition with char variable&comparison expression
		int b = pos(8); //function call and assignment command
		int num;
		scanf(num); //scanf function called in while
	}
}

File: LanguageTranslator/tests/Python/while.py
# /*
#  * University of Glasgow
#  * Msc Project fall, 2019
#  * Author: Yidi Cao
#  * Acknowledgement:
#  * This file is partly based on the Fun Compiler at https://moodle.gla.ac.uk/course/view.php?id=903
#  * 
# */
# Test while-else commands
a = input()
i = 1
def fac(a,b):
	i = b+1+a
	return i
while i*i < a+1: #comparison expression as condition
	a = fac(a,9) #function called in while statement
	if a>10: #if-elif-else statement in while statement
		print(i*i)
	elif a == 10:
		print(a)
	else:
		print(a+1)
	i = i+1
else:
    def pos(a): #function declared in while statement
		return a+i
	print(pos(8))

while fac(i,a): #function call as condition
	print(a)